modules/shapefile/src/lib/parsers/parse-shapefile.js

```diff
@@ -1,10 +1,12 @@
-import {binaryToGeoJson} from '@loaders.gl/gis';
+import {Proj4Projection} from '@math.gl/proj4';
+import {binaryToGeoJson, transformGeoJsonCoords} from '@loaders.gl/gis';
 import {parseShx} from './parse-shx';
 import {zipBatchIterators} from '../streaming/zip-batch-iterators';
 import {SHPLoader} from '../../shp-loader';
 import {DBFLoader} from '../../dbf-loader';
 
 export async function* parseShapefileInBatches(asyncIterator, options, context) {
+  const {_targetCrs = 'WGS84'} = (options && options.gis) || {};
```

Comment:
Did you want to support `options.gis.reproject`?

Reply:
I didn't realize that was a standard option

Reply:
Not quite sure what you are saying, but this is how I think:

- `options.gis` are category options, while `options.shapefile` are loader specific options.
- so if we add an option to `options.gis`, our intention is presumbly that it can apply to multiple geospatial loaders.
- so, are there geospatial formats other than shapefile that can specify CRS? 
- I think the answer is yes. I have for instance come across geojson files that have a (non-standard) `crs` field.

Reply:
> * so if we add an option to `options.gis`, our intention is presumbly that it can apply to multiple geospatial loaders.

Yes, that is my intention.

> * so, are there geospatial formats other than shapefile that can specify CRS?

Yes, of the existing loaders, WKB and Flatgeobuf can specify CRS. GeoPackage, a possible future loader, also can specify CRS.

> * I have for instance come across geojson files that have a (non-standard) `crs` field

The older [GeoJSON spec from 2008](https://geojson.org/geojson-spec.html#coordinate-reference-system-objects) did allow a CRS field. The newer GeoJSON spec from 2016 [removed CRS support](https://tools.ietf.org/html/rfc7946#section-4) from the specification, and all coordinates must be in WGS84.

Reply:
Of other formats, Vector Tiles have their localized coordinate system; I believe WKT is supposed to be in WGS84 but not entirely sure.

Reply:
Awesome, I love your encyclopaedic knowledge on geospatial formats! And I think I3S can also specify a CRS.

As a side note, I wouldn't mind supporting the older geojson standard, though the files I have encountered had the `crs` field at the end of the file, after the features, meaning the streaming JSON parser would have already yielded the batches when it would realize that it should have reprojected them first (duh)...

Reply:
BTW, can we handle these CRS specifiers (from geojson-2008 files) `"name": "urn:ogc:def:crs:OGC:1.3:CRS84"`

Reply:
> though the files I have encountered had the `crs` field at the end of the file, after the features

I assume the JSON spec doesn't specify any ordering of keys, so there's no way to know ahead of time if the CRS field doesn't exist, or exists after the features.

> BTW, can we handle these CRS specifiers (from geojson-2008 files) `"name": "urn:ogc:def:crs:OGC:1.3:CRS84"`

It looks like there are two options from that spec:

1. Linked objects:

```json
  "crs": {
    "type": "link", 
    "properties": {
      "href": "http://example.com/crs/42",
      "type": "proj4"
      }
    }
```
this seems doable since we'd just have to fetch that `href` and pass it to Proj4.js.

2. Named CRS:

```json
  "crs": {
    "type": "name",
    "properties": {
      "name": "urn:ogc:def:crs:OGC:1.3:CRS84"
      }
    }
```
This relies on understanding all the OGC CRS identifiers. Since Proj4.js only includes a few named projections, we'd have to try to deduce the CRS from the name somehow or ping a service like epsg.io.

I don't think it's worth the hassle at this point, unless we get feedback from loaders.gl users that they have data in this older spec.